bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -56,39 +58,81 @@ export class StringFormatter extends CellFormatter {
   }
 
   static {
-    this.define<StringFormatter.Props>(({Color, Nullable, String}) => ({
-      font_style: [ FontStyle, "normal" ],
-      text_align: [ TextAlign, "left"   ],
-      text_color: [ Nullable(Color), null ],
+    this.define<StringFormatter.Props>(({String}) => ({
+      font_style: [ p.FontStyleSpec, {value: "normal"} ],
+      text_align: [ p.TextAlignSpec, {value: "left"} ],
+      text_color: [ p.ColorSpec, null ],
+      background_color: [ p.ColorSpec, null ],
       nan_format: [ String, "-"],
     }))
   }
 
   override doFormat(_row: any, _cell: any, value: any, _columnDef: any, _dataContext: any): string {
-    const {font_style, text_align, text_color} = this
+    const {font_style, text_align, text_color, background_color} = this
 
     const text = div(value == null ? "" : `${value}`)
-    switch (font_style) {
+
+    // Font style
+    let resolvedFontStyle
+    if (isValue(font_style)) {
+      resolvedFontStyle = font_style.value
+    } else if (isField(font_style)) {
+      resolvedFontStyle = _dataContext[font_style.field]
+    }
```

Comment:
This is unfortunate, because this should be deferred to vectorized properties to figure out how to resolve this. Especially to handle ndarrays properly, among other issues. Having access to the data source, we could write something like this: 
```ts
this.properties.font_style.uniform(data_source).get(index)
```
though this would need to be precomputed, perhaps as some kind of virtual column in table's data provider.

Reply:
I will probably need guidance on this - not sure where/when to precompute. Are we doing something similar somewhere else in the code I can use for inspiration ?

Reply:
@mattpap is this something we could merge as-is for now with a follow-up issue to improve later?

Reply:
This is definitely not ideal and compiles only because it's mostly dynamically typed code, but given the proper solution is non-trivial and that this isn't critical code path in bokehjs, then yes we can proceed with this and improve later.